### PR TITLE
LADX: more item groups, location groups, keysanity preset

### PR DESCRIPTION
**Pull request by threeandthreee**
## What is this fixing or adding?
Adds a bunch more item groups, creates location groups by area, and adds a keysanity preset to the options page.

## How was this tested?
Generated some seeds, tested preset in webhost, checked datapackage to see that groups were made correctly.

### CODE REVIEW

File: worlds/ladx/Items.py [hunk 1]
```diff
@@ -305,3 +305,65 @@ class ItemName:
...
+    "Upgrades": {
+        "Max Powder Upgrade", "Max Bombs Upgrade", "Max Arrows Upgrade"
+    },
+    "Songs": {
+        "Ballad of the Wind Fish", "Manbo's Mambo", "Frog's Song of Soul"
+    },
+    "Tunics": {
+        "Red Tunic", "Blue Tunic"
+    }
+}
```
Comment (Exempt-Medic) [inline worlds/ladx/Items.py:369]: Need an end-of-file newline here. Also, the sets here are inconsistently formatted, some of them put two items per "row" other put four, or six, or just every item. It's also inconsistent whether or not these lists end in commas, I'd suggest just making them all one item per row and including commas after the last item.
```suggestion
}

```

File: worlds/ladx/__init__.py [hunk 2]
```diff
@@ -66,6 +68,15 @@ class LinksAwakeningWebWorld(WebWorld):
     )]
     theme = "dirt"
     option_groups = ladx_option_groups
+    options_presets = {
+        "Keysanity": {
```
Comment (Exempt-Medic) [inline worlds/ladx/__init__.py:72]: I would probably change this, maps, compasses, and stone beaks are not very "key" related. You could add in Instruments and call it "All Shuffle" or something maybe?

Comment (ScipioWright) [reply worlds/ladx/__init__.py:72]: Well, I think upstream's preset is poorly named, then. "Shuffle Dungeon Items" maybe?

File: worlds/ladx/Locations.py [hunk 3]
```diff
@@ -25,6 +25,17 @@
...
 
+def get_location_name_groups():
+    ret = {}
+    for s, v in checkMetadataTable.items():
+        if s == "None":
+            continue
+        if v.area not in ret:
+            ret[v.area] = []
+        ret[v.area].append(meta_to_name(v))
+    return ret
```
Comment (ScipioWright) [inline worlds/ladx/Locations.py:36]:
```suggestion
def get_location_name_groups() -> Dict[str, Set[str]]:
    ret = {}
    for s, v in checkMetadataTable.items():
        if s == "None":
            continue
        ret.setdefault(v.area, set()).add(meta_to_name(v))
    return ret
```
It's supposed to be a Dict[str, Set[str]], not a Dict[str, List[str]]. Import `Dict` and `Set` from `typing` too.

`dict.setdefault` does the same thing as checking if something is in the dict and, if it's not, giving it a key and setting a default value.

File: worlds/ladx/Items.py [hunk 4]
```diff
@@ -305,3 +305,120 @@ class ItemName:
 links_awakening_items_by_name = {
     item.item_name : item for item in links_awakening_items
 }
+
+links_awakening_item_name_groups = {
```
Comment (ScipioWright) [inline worlds/ladx/Items.py:309]:
```suggestion
links_awakening_item_name_groups: Dict[str, Set[str]] = {
```
And also import `Dict` and `Set` from `typing`

File: worlds/ladx/__init__.py [hunk 5]
```diff
@@ -66,6 +68,15 @@ class LinksAwakeningWebWorld(WebWorld):
     )]
     theme = "dirt"
     option_groups = ladx_option_groups
+    options_presets = {
```
Comment (ScipioWright) [inline worlds/ladx/__init__.py:71]:
```suggestion
    options_presets: typing.Dict[str, typing.Dict[str, typing.Any]] = {
```

**Review by ScipioWright — COMMENTED** [review]
Some cleanup, and some typing. While the rest of the world doesn't have good typing, that doesn't mean we shouldn't add some when new things are added.

**Review by SushiKishi — APPROVED** [review]
More logic options are always fun. Seeds with this implemented have generated as expected.

**Review by palex00 — APPROVED** [review]
This PR was tested as part of a Beta apworld that was used in three dedicated Test Asyncs, as well as dozen of normal Syncs. I have also personally run 200 test generations which succeeded.

The new feature has worked as expected and other stuff does not seem impacted.

Also let's be real, this is a very simple code change. It's just some design decision semantics.

File: worlds/ladx/__init__.py (hunk 2 above)
Comment (NewSoupVi) [reply worlds/ladx/__init__.py:72]: I agree with the points made by Medic and Scipio, HOWEVER: At the moment, presets always change every option. As a result, I think it's fine to add other options that seem to have a similar "spirit"

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**